lib/validator/validator.go

```diff
@@ -36,22 +36,28 @@ import (
 var validate *validator.Validate
 
 var (
-	labelRegex          = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `$`)
-	labelValueRegex     = regexp.MustCompile("^[a-zA-Z0-9]?([a-zA-Z0-9_.-]{0,61}[a-zA-Z0-9])?$")
-	nameRegex           = regexp.MustCompile("^[a-zA-Z0-9_.-]{1,128}$")
-	namespacedNameRegex = regexp.MustCompile(`^[a-zA-Z0-9_./-]{1,128}$`)
-	interfaceRegex      = regexp.MustCompile("^[a-zA-Z0-9_-]{1,15}$")
-	actionRegex         = regexp.MustCompile("^(allow|deny|log|pass)$")
-	backendActionRegex  = regexp.MustCompile("^(allow|deny|log|next-tier|)$")
-	protocolRegex       = regexp.MustCompile("^(tcp|udp|icmp|icmpv6|sctp|udplite)$")
-	ipipModeRegex       = regexp.MustCompile("^(always|cross-subnet|)$")
-	reasonString        = "Reason: "
-	poolSmallIPv4       = "IP pool size is too small (min /26) for use with Calico IPAM"
-	poolSmallIPv6       = "IP pool size is too small (min /122) for use with Calico IPAM"
-	poolUnstictCIDR     = "IP pool CIDR is not strictly masked"
-	overlapsV4LinkLocal = "IP pool range overlaps with IPv4 Link Local range 169.254.0.0/16"
-	overlapsV6LinkLocal = "IP pool range overlaps with IPv6 Link Local range fe80::/10"
-	protocolPortsMsg    = "rules that specify ports must set protocol to TCP or UDP"
+	labelRegex            = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `$`)
```

Comment:
The label regexes should exactly match those in kubernetes now, rather than the one in the tokenizer code.

Reply:
I found what looks like the k8s regex for labels in apimachinery: https://goo.gl/Sv4h6t

We shouldn't try to replicate what they're doing at this point since the approach is significantly different. How about removing reference to the `tokenizer` and having a standalone regex?

Reply:
yup  - that's what i was trying to suggest :-)  Key thing is that we only allow the same names/labels etc.

Reply:
`labelValueRegex` evaluates identically to apimachinery's `IsQualifiedName()`. I've removed the dependency on `tokenizer`.